specification/ai/ModelClient/routes.tsp

```diff
@@ -0,0 +1,71 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+
+import "./models/common.tsp";
+import "./models/chat_completions.tsp";
+import "./models/embeddings.tsp";
+import "./models/image_embeddings.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+namespace AI.Model;
+
+@doc("""
+  Gets chat completions for the provided chat messages.
+  Completions support a wide variety of tasks and generate text that continues from or "completes"
+  provided prompt data. The method makes a REST API call to the `/chat/completions` route 
+  on the given endpoint.
+  """)
+@actionSeparator("/")
+@route("chat/completions")
+op getChatCompletions is Azure.Core.RpcOperation<
+  {
+    ...ChatCompletionsOptions;
+    ...AdditionalRequestHeaders;
+  },
+  ChatCompletions
+>;
+
+@doc("""
+  Return the embedding vectors for given text prompts.
+  The method makes a REST API call to the `/embeddings` route on the given endpoint.
+  """)
+@actionSeparator("/")
+@route("embeddings")
+op getEmbeddings is Azure.Core.RpcOperation<
+  {
+    ...EmbeddingsOptions;
+    ...AdditionalRequestHeaders;
+  },
+  EmbeddingsResult
+>;
+
+@doc("""
+  Return the embedding vectors for given images.
+  The method makes a REST API call to the `/images/embeddings` route on the given endpoint.
+  """)
+@actionSeparator("/")
+@route("images/embeddings")
+op getImageEmbeddings is Azure.Core.RpcOperation<
+  {
+    ...ImageEmbeddingsOptions;
+    ...AdditionalRequestHeaders;
+  },
+  EmbeddingsResult
+>;
```

Comment:
@dargilco 

Please check if you can put the `###Options` to the explicit `@body`, e.g.

```ts
op getChatCompletions is Azure.Core.RpcOperation<
  {
    @body body: ChatCompletionsOptions;
    ...AdditionalRequestHeaders;
  },
  ChatCompletions
>;
```

The reason is that without `@body` would let SDK emitter spread the body properties to method signature. (see [spread](https://azure.github.io/typespec-azure/docs/howtos/Client%20Generation/08methodInputs#spread)).

However, SDK emitter cannot handle the `...Record<unknown>` in the `###Options` in above scenario. -- `...Record<unknown>` means any key-value can be set to the body request, but that won't work on the method signature.

Therefore, this is going to cause problem to SDK emitters, and the output code would be incorrect.

Reply:
Basically, the request body is properties
```
messages
frequency_penalty
...
<any undocumented properties form Record>
```

But after the spread in method (the way of writing the alias `ChatCompletionsOptions` in `getChatCompletions` without `@body`), the SDK method signature would become
`getChatCompletions(messages, frequency_penalty, ...)` without a way to specify undocumented properties.

Reply:
The error reported by TCGC would make Java emitter fail.

```
error @azure-tools/typespec-client-generator-core/no-corresponding-method-param: Missing "completeRequest" method parameter in method "complete", when "completeRequest" must be sent to the service. Add a parameter named "completeRequest" to the method.
error @azure-tools/typespec-client-generator-core/no-corresponding-method-param: Missing "embedRequest" method parameter in method "embed", when "embedRequest" must be sent to the service. Add a parameter named "embedRequest" to the method.
error @azure-tools/typespec-client-generator-core/no-corresponding-method-param: Missing "embedRequest1" method parameter in method "embed", when "embedRequest1" must be sent to the service. Add a parameter named "embedRequest1" to the method.
F
```